crates/re_ui/src/drag_and_drop.rs

```diff
@@ -0,0 +1,320 @@
+//! Helpers for drag and drop support for reordering hierarchical lists.
```

Comment:
I really like that this is a completely separate module, with no deps. We could consider moving it to `egui`; then it could be used for https://github.com/emilk/egui/pull/3903 as well

Reply:
Yes, that's kind of what I had in mind. I would wait a little bit though and see how it crystallise in the short/mid term.